runtime/src/tests.rs

```diff
@@ -173,37 +189,72 @@ fn pallet_poe_availability() {
 mod pallets_interact {
     use super::*;
 
-    #[test]
-    fn session_notifies_staking() {
-        new_test_ext().execute_with(|| {
-            let pre_staking_session = Staking::current_planned_session();
-            Session::rotate_session();
-            let post_staking_session = Staking::current_planned_session();
-            assert_eq!(pre_staking_session + 1, post_staking_session);
-        });
+    // Any random values for these constants should do
+    const BLOCK_NUMBER: BlockNumber = 1;
+    const SLOT_ID: u64 = 87;
+    const BABE_AUTHOR_ID: u32 = 1;
+
+    // Initialize block #BLOCK_NUMBER, authored at slot SLOT_ID by BABE_AUTHOR_ID using Babe
+    fn initialize() {
+        let slot = Slot::from(SLOT_ID);
+        let authority_index = BABE_AUTHOR_ID;
+        let transcript = sp_consensus_babe::VrfTranscript::new(b"test", &[]);
```

Comment:
You're using this function to init the chain in almost all following tests. Consider to redefine `new_test_ex()` function to include this code too:

```rust
    fn new_test_ext() -> sp_io::TestExternalities {
        let mut ex = super::new_test_ext();
        ex.execute_with(|| initialize());
        ex
    }
```

Reply:
Good idea, done!